src/test/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetTransferServiceImplTest.java

```diff
@@ -0,0 +1,251 @@
+package edu.cornell.kfs.module.cam.document.service.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.regex.PatternSyntaxException;
+
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+import org.kuali.kfs.module.cam.CamsConstants;
+import org.kuali.kfs.module.cam.businessobject.Asset;
+import org.kuali.kfs.module.cam.businessobject.AssetGlpeSourceDetail;
+import org.kuali.kfs.module.cam.document.service.impl.AssetLocationServiceImpl;
+import org.kuali.kfs.module.cam.document.service.impl.AssetObjectCodeServiceImpl;
+import org.kuali.kfs.module.cam.document.service.impl.AssetPaymentServiceImpl;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.service.impl.KfsParameterConstants;
+import org.kuali.kfs.sys.service.impl.UniversityDateServiceImpl;
+import org.kuali.rice.core.impl.datetime.DateTimeServiceImpl;
+import org.kuali.rice.coreservice.api.parameter.EvaluationOperator;
+import org.kuali.rice.coreservice.api.parameter.Parameter;
+import org.kuali.rice.coreservice.api.parameter.ParameterType;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.coreservice.impl.parameter.ParameterServiceImpl;
+import org.kuali.rice.krad.service.impl.BusinessObjectServiceImpl;
+
+import edu.cornell.kfs.module.cam.CuCamsConstants;
+
+public class CuAssetTransferServiceImplTest {
+
+    private static final String TEST_ACCOUNT = "G104700";
+    private static final String TEST_SUB_ACCOUNT = "TECH";
+    private static final String TEST_PLANT_ACCOUNT = "G10CAPT";
+
+    private static final String TEST_PATTERN_WILDCARD = "*CAPT";
+    private static final String TEST_PATTERN_REGULAR_EXACT_MATCH = TEST_ACCOUNT;
+    private static final String TEST_PATTERN_PLANT_EXACT_MATCH = TEST_PLANT_ACCOUNT;
+    private static final String TEST_PATTERN_WILDCARD_BOTH_MATCH = "G10*";
+    private static final String TEST_PATTERN_NO_MATCH = "NONE";
+    private static final String TEST_MULTI_PATTERN = "*XXXX;*CAPT";
+    private static final String TEST_MULTI_PATTERN_REGULAR_MATCH = "*XXXX;G104*";
+    private static final String TEST_MULTI_PATTERN_NO_MATCH = "*XXXX;NONE";
+    private static final String TEST_BAD_PATTERN = "[*CAPT}";
+
+    // These flags are for readability convenience.

```

Comment:
Not sure this and some of the other comments in this test are adding much value.